receiver/receivercreator/discovery.go

```diff
@@ -0,0 +1,185 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package receivercreator // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/receivercreator"
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/go-viper/mapstructure/v2"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v3"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer"
+)
+
+const (
+	// hints prefix
+	otelHints = "io.opentelemetry.discovery"
+
+	// hint suffix for metrics
+	otelMetricsHints = otelHints + ".metrics"
+
+	// hints definitions
+	discoveryEnabledHint = "enabled"
+	scraperHint          = "scraper"
+	configHint           = "config"
+)
+
+// hintsTemplatesBuilder creates configuration templates from provided hints.
+type hintsTemplatesBuilder interface {
+	createReceiverTemplateFromHints(env observer.EndpointEnv) (*receiverTemplate, error)
+}
+
+// k8sHintsBuilder creates configurations from hints provided as Pod's annotations.
+type k8sHintsBuilder struct {
+	logger          *zap.Logger
+	config          DiscoveryConfig
+	ignoreReceivers map[string]bool
+}
+
+var _ hintsTemplatesBuilder = (*k8sHintsBuilder)(nil)
+
+func createK8sHintsBuilder(config DiscoveryConfig, logger *zap.Logger) k8sHintsBuilder {
+	ignoreReceivers := make(map[string]bool, len(config.IgnoreReceivers))
+	for _, r := range config.IgnoreReceivers {
+		ignoreReceivers[r] = true
+	}
+	return k8sHintsBuilder{
+		logger:          logger,
+		config:          config,
```

Comment:
Why do we need to store the full config if we have a separate field for `ignoreReceivers`? Can be just enabled right?

Reply:
Must be a leftover after some of the re-designs, good catch! Remove that now, thank's!